10-setup-htcondor.sh

```diff
@@ -228,23 +225,66 @@ EOF
 
 fi
 
-if is_true "$ITBPOOL"; then
-    OSPOOL_CM1=cm-1.ospool-itb.osg-htc.org
-    OSPOOL_CM2=cm-2.ospool-itb.osg-htc.org
+
+######################
+# POOL CONFIGURATION #
+######################
+
+# Default to the production OSPool
+POOL=${POOL:=prod-ospool}
+
+case ${POOL} in
+    itb-ospool)
+        default_cm1=cm-1.ospool-itb.osg-htc.org
+        default_cm2=cm-2.ospool-itb.osg-htc.org
+        default_syslog_host=syslog.osgdev.chtc.io
+        ;;
+    prod-ospool)
+        default_cm1=cm-1.ospool.osg-htc.org
+        default_cm2=cm-2.ospool.osg-htc.org
+        default_syslog_host=syslog.osg.chtc.io
+        ;;
+    prod-path-facility)
+        default_cm1=cm-1.facility.path-cc.io
+        default_cm2=cm-2.facility.path-cc.io
+        default_syslog_host=syslog.osg.chtc.io
+        ;;
+    *)
+        echo "Unknown pool $POOL" >&2
+        exit 1
+        ;;
+esac
+
+# Allow users to override the pool configuration
+if [[ -n $CONDOR_HOST ]]; then
+    # if the user sets $CONDOR_HOST, we can't assume the POOL
+    # so we unset it here to avoid confusion
+    POOL=
 else
-    OSPOOL_CM1=cm-1.ospool.osg-htc.org
-    OSPOOL_CM2=cm-2.ospool.osg-htc.org
+    CONDOR_HOST=$default_cm1,$default_cm2
 fi
 
-# extra HTCondor config
-# if CCB_RANGE_* is set, use the old config, otherwise assume OSPool with shared port
-if [[ "x$CCB_RANGE_LOW" != "x" ]]; then
-    CCB_PORT=$(python -S -c "import random; print(random.randrange($CCB_RANGE_LOW,$CCB_RANGE_HIGH+1))")
-    CCB_ADDRESS="\$(CONDOR_HOST):$CCB_PORT"
-else
-    CCB_COLLECTOR=$(python -S -c "import random; print(random.randrange(1,6))")
-    CCB_ADDRESS="${OSPOOL_CM1}:9619?sock=collector$CCB_COLLECTOR,${OSPOOL_CM2}:9619?sock=collector$CCB_COLLECTOR"
+# Configure remote peer if applicable
+SYSLOG_HOST=${SYSLOG_HOST:-$default_syslog_host}
+
+if [[ -n $CCB_RANGE_LOW && $CCB_RANGE_HIGH ]]; then
+    # Choose a random CCB port if the user gives us a port range
+    # e.g., cm.school.edu:10576
+    CCB_SUFFIX=$(random_range "$CCB_RANGE_LOW" "$CCB_RANGE_HIGH")
+elif [[ $POOL =~ (itb|prod)-ospool ]]; then
+    # Choose a random OSPool collector for CCB
+    # e.g., cm-1.ospool.osg-htc.org:9619?sock=collector3
+    CCB_SUFFIX="9619?sock=collector$(random_range 1 6)"
 fi
+
+# Append the CCB suffix to each host in CONDOR_HOST
+if [[ -n $CCB_RANGE_LOW && $CCB_RANGE_HIGH ]] ||
+       [[ $POOL =~ (itb|prod)-ospool ]]; then
+    CCB_ADDRESS=$(python -Sc "import re; \
+print(','.join([cm + ':$CCB_SUFFIX' \
+for cm in re.split(r'[\s,]+', '$CONDOR_HOST')]))")
```

Comment:
Can you add an example of what this looks like?

Reply:
Done!

Reply:
Thanks!